nptdms/tdms_segment.py

```diff
@@ -420,6 +440,18 @@ def _have_interleaved_data(self):
         else:
             raise ValueError("Cannot read interleaved segment containing channels with unsized types")
 
+    def _get_data_objects(self):
+        if self.data_objects_cached is not None:
+            return self.data_objects_cached
+
+        self.data_objects_cached = [o for o in self.ordered_objects if o.has_data]
+        return self.data_objects_cached
+
+    def _invalidate_cached_values(self):
```

Comment:
I don't think this cache invalidation is needed, the cached properties won't have been accessed at the places where this is currently called

Reply:
Ok nice, i didn't see the full picture yet. Tests are passing without :)